### PR TITLE
#5569. shareUrl parsing fixed

### PR DESCRIPTION
**Pull request by jstrR**
## Description

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix

## Issue

**What is the current behavior?**

#5569 https://github.com/geosolutions-it/mapstore2/issues/5569#issuecomment-661852246

**What is the new behavior?**

Fixes https://github.com/geosolutions-it/mapstore2/issues/5569#issuecomment-661852246
## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/components/share/SharePanel.jsx [hunk 1]
```diff
@@ -168,9 +168,9 @@ class SharePanel extends React.Component {
             shareUrl = `${shareUrl}${settings.markerEnabled ? "?marker=" : "?center="}${this.state.coordinate}&zoom=${this.state.zoom}`;
         }
         if (!settings.showSectionId && advancedSettings && advancedSettings.sectionId) {
-            const parsedUrl = shareUrl.split('/');
-            if (parsedUrl.length === 8) shareUrl = replace(shareUrl, parsedUrl[parsedUrl.length - 1], '');
-            if (parsedUrl.length === 9) shareUrl = replace(shareUrl, `${parsedUrl[parsedUrl.length - 2]}/${parsedUrl[parsedUrl.length - 1]}`, '');
+            const parsedUrl = shareUrl.split('#')[1]?.split('/');
+            if (parsedUrl.length === 5 && parsedUrl.includes('shared')) shareUrl = replace(shareUrl, parsedUrl[parsedUrl.length - 1], '');
+            if (parsedUrl.length === 6 && parsedUrl.includes('shared')) shareUrl = replace(shareUrl, `${parsedUrl[parsedUrl.length - 2]}/${parsedUrl[parsedUrl.length - 1]}`, '');
```
Comment (allyoucanmap) [inline web/client/components/share/SharePanel.jsx:173]: The issue with the previus code was about the different path in the url between the local and the test enviroments. Now this changes ensure that the shareUrl is split on the hash before count the paresd parts of the path. This solutions seems to work but I would like to move this logic inside the [getSharedGeostoryUrl](https://github.com/geosolutions-it/MapStore2/blob/4ee11de7be140b78888fa304b6992a76e8feb5e2/web/client/utils/ShareUtils.js#L64-L68) function used some lines above https://github.com/geosolutions-it/MapStore2/blob/4ee11de7be140b78888fa304b6992a76e8feb5e2/web/client/components/share/SharePanel.jsx#L117

this is an example I did not test the code

in  ShareUtils.js
```js
getSharedGeostoryUrl: (url = '', removeSectionId) => {
    // check if the url is from geostory page
   // bacause share is used also in other page (eg: map viewer, dashboard)
    if (url.match(/\#\/(geostory)/)) {
        let geostoryUrl = url.match(/\/(geostory)\/((shared)|(newgeostory))/)
            ? url
            : url.replace('/geostory/', '/geostory/shared/');
        
        if (removeSectionId) {
            // implementation
        }
        return geostoryUrl;
    }

    return url;
}
```

in SharePanel.jsx

```js
const removeSectionId = !settings.showSectionId && advancedSettings && advancedSettings.sectionId;
let shareUrl = getSharedGeostoryUrl(removeQueryFromUrl(this.props.shareUrl), removeSectionId);

```

in this way we have all the logic for the geostory url in one function and we can add some tests.

File: web/client/epics/geostory.js [hunk 2]
```diff
@@ -283,7 +284,19 @@ export const loadGeostoryEpic = (action$, {getState = () => {}}) => action$
                 return axios.get(`configs/${id}.json`)
                     // not return anything else that data in this case
                     // to match with data/resource object structure of getResource
-                    .then(({data}) => ({ data, isStatic: true, canEdit: true }));
+                    .then(({data}) => {
+                        // generates random id for section
+                        const defaultSection = data.sections[0];
```
Comment (allyoucanmap) [inline web/client/epics/geostory.js:289]: This section id replacement is specific for [newgeostory.json](https://github.com/geosolutions-it/MapStore2/blob/4ee11de7be140b78888fa304b6992a76e8feb5e2/web/client/configs/newgeostory.json) configuration but the string `id` could be different from `newgeostory` and it contain a diffferent structure of the configuration. could we add a control to add the uuid only if the `id` is equal to `newgeostory`?
What do you think @offtherailz ?

File: web/client/utils/GeoStoryUtils.js [hunk 3]
```diff
@@ -454,7 +454,7 @@ export const parseHashUrlScrollUpdate = (url, hash = '', storyId, sectionId, col
             return replace(url, `${storyIds[1]}`, `${sectionId}`);
         }
         if (hash.includes('shared')) {
-            return `${url}${sectionId}`;
+            return `${url}/${sectionId}`;
```
Comment (allyoucanmap) [inline web/client/utils/GeoStoryUtils.js:457]: While testing other functionality I removed manually the section id from the url so I got the path with the trailing slash `#/geostory/:gid/` this caused a wrong parsing of the path stacking new id on the same path.

see the url in this gif
trailing-slash-1

I'm not sure the issue is related on this function anyway if the url argument is expected without trailing slash we should ensure is it removed from it in this parsing function before to add new parts to the path.

**Comment by offtherailz** [comment]
please @jstrR provide a backport for 2020.02.xx

**Pull request merged**